lxd/cluster/connect.go

```diff
@@ -162,29 +164,30 @@ func ConnectIfVolumeIsRemote(s *state.State, poolName string, projectName string
 				return nil, errors.Wrapf(err, "Failed getting cluster member info for %q", remoteInstance.Node)
 			}
 
-			// Replace address list with instance's cluster member.
-			addresses = []string{instNode.Address}
+			// Replace node list with instance's cluster member node (which might be local member).
+			nodes = []db.NodeInfo{instNode}
 		} else {
-			// Volume isn't exclusively attached to an instance and has no fixed node.
-			addresses = []string{""} // Use local cluster member.
+			// Volume isn't exclusively attached to an instance. Use local cluster member.
+			return nil, nil
 		}
 	}
 
-	addressCount := len(addresses)
-	if addressCount > 1 {
-		return nil, fmt.Errorf("More than one cluster member has a volume named %q", volumeName)
-	} else if addressCount < 1 {
+	nodeCount := len(nodes)
+	if nodeCount > 1 {
+		return nil, fmt.Errorf("More than one cluster member has a volume named %q. Use --target flag to specify member", volumeName)
```

Comment:
Not worth fixing in this one (I'd rather have it land now) but in general we should not include CLI flags in server errors as there's no telling what the client may actually be. So Something more generic like "Please target a specific member" would be an option here.

Reply:
@stgraber ok no problem I'll fix in the next RP.